### PR TITLE
[5.6] Fix SQLite delete for complex delete statements

### PR DESCRIPTION
**Pull request by abdumu**
Current SQLite delete implementation lack the support for complex statement like joins, or limits. This may lead to serious loss of data.

Current approach of SQLite delete: 

`> App\Post::latest('id')->limit(1)->delete();`
`=> 48 (deleted rows)`

`=> "query" => "delete from "posts""`

new approach:

`> App\Post::latest('id')->limit(1)->delete();`
`=> 1 (deleted rows)`

`=>  "query" => "delete from "posts" where "rowid" in (select "posts"."rowid" from "posts" order by "id" desc limit 1)"`

Also, this fixes the https://github.com/laravel/framework/pull/22239 for SQLite.

### CODE REVIEW

File: src/Illuminate/Database/Query/Grammars/SQLiteGrammar.php [hunk 1]
```diff
@@ -174,6 +175,48 @@ public function compileInsert(Builder $query, array $values)
...
+    /**
+     * Compile a delete statement into SQL.
+     *
+     * @param  \Illuminate\Database\Query\Builder  $query
+     * @return string
+     */
+    public function compileDelete(Builder $query)
+    {
+        // SQLite delete statment doesn't fully support complex keywords like joins ..
+        // We use select statment as a subquery to overcome this delimma.
```
Comment (deleugpn) [inline src/Illuminate/Database/Query/Grammars/SQLiteGrammar.php]: Typo

**Comment by deleugpn** [comment]
Cool solution.

**Comment by themsaid** [comment]
Since it's sqlite it's pretty easy to add integration tests, please add a test in `tests/Integration/Database` with the cases this PR fixes.

**Review by themsaid — CHANGES_REQUESTED** [review]
Since it's sqlite it's pretty easy to add integration tests, please add a test in tests/Integration/Database with the cases this PR fixes.

File: src/Illuminate/Database/Query/Grammars/SQLiteGrammar.php [hunk 2]
```diff
@@ -174,6 +175,48 @@ public function compileInsert(Builder $query, array $values)
...
+    public function compileDelete(Builder $query)
+    {
+        // SQLite delete statement doesn't fully support complex keywords like joins ..
+        // We use select statement as a sub-query to overcome this dilemma.
+        if (isset($query->joins) || isset($query->limit)) {
+            // Since rowid is common column between all SQLite tables,
+            // we use it in select sub-query and in delete where-in statement.
+            $selectSql = parent::compileSelect($query->select("{$query->from}.rowid"));
+
+            return trim(
```
Comment (ntzm) [inline src/Illuminate/Database/Query/Grammars/SQLiteGrammar.php]: I don't think this `trim` is needed

**Pull request merged**